src/assets/scripts/app.js

```diff
@@ -1,23 +1,72 @@
 "use strict";
 
-// Menu
-jQuery("#menu").menu({
-    position: {my: "right top", at: "right bottom"},
-    icons: {submenu: "ui-icon-blank"},
-    // Remove the anchors from the tabindex as they are triggered
-    // via the select event. (see below)
-    create: function () {
-        jQuery("#menu .ui-menu-item a").attr("tabindex", -1);
-    },
-    // The jQuery menu doesn't seem to like it when the <a> are
-    // specified as the "item" and are nested under an <li>. There
-    // seems to be styling issues and occassionally an error thrown.
-    // The page transition work around below is used to maintian the semantic structure,
-    // while allowing the menu to behave as normal.
-    select: function (event, ui) {
-        var location = ui.item.children("a").attr("href");
-        if (location) {
-            window.location.href = location;
-        }
+const menu = {};
+
+/**
+ * Toggle's the aria-expanded state of an element.
+ *
+ * @param {DOMNode} element - the DOM Node to toggle the aria-expanded state on.
+ * @param {Boolean} [state] - (optional) explicit state to set aria-expanded to.
+ */
+menu.toggleExpansion = (element, state) => {
+    let isExpanded = element.getAttribute("aria-expanded") === "true" ? true : false;
+    state = typeof(state) !== "undefined" ? state : !isExpanded;
+
+    element.setAttribute("aria-expanded", state ? "true" : "false");
+};
+
+/**
+ * Initializes a menu widget.
+ *
+ * @param {String} container - A CSS selector for the menu's container.
+ * @param {String} [button] - (optional) A CSS slector for the menu's button. Defaults to "button".
+ * @throws {Exception} - if no element is found for the container selector.
+ * @throws {Exception} - if no element is found for the button selector within the container element.
+ */
+menu.init = (container, button = "button") => {
+    let menuContainer = document.querySelector(container);
+
+    if (!menuContainer) {
+        throw new Error(`No element found for container selector: ${container}`);
     }
-});
+
+    let btn = menuContainer.querySelector(button);
+
+    if (!btn) {
+        throw new Error(`No element found for button selector: ${button}`);
+    }
+
+    // Close the menu when focus is moved away from the menu
+    menuContainer.addEventListener("focusout", (event) => {
+        if (!menuContainer.contains(event.relatedTarget)) {
+            menu.toggleExpansion(btn, false);
+        }
+    });
+
+    // Close the menu when mouseleave fires. This is needed for Safari on iOS which doesn't fire the focusout event
+    // when touching another part of the page.
+    menuContainer.addEventListener("mouseleave", () => {
+        if (!menuContainer.contains(document.activeElement)) {
+            menu.toggleExpansion(btn, false);
+        }
+    });
```

Comment:
Use the following to address the issues with the menu in Safari. This should work in iOS and macOS. However, it isn't completely matching other browsers. For example if you right click outside of the menu it will stay open, and clicking in an iframe also won't close the menu. These seemed more minor though.
```suggestion
    // Close the menu when clicking outside of it.
    // This is needed due to Safari not applying focus to clicked inputs.
    // https://bugs.webkit.org/show_bug.cgi?id=22261
    document.body.addEventListener("click", (event) => {
        if (!menuContainer.contains(event.target)) {
            menu.toggleExpansion(btn, false);
        }
    });
```